stage1/usr_from_kvm/lkvm/patches/dead_lock_fix.patch

```diff
@@ -0,0 +1,35 @@
+diff --git a/builtin-run.c b/builtin-run.c
+index edcaf3e..cb539a4 100644
+--- a/builtin-run.c
++++ b/builtin-run.c
+@@ -630,7 +630,7 @@ static struct kvm *kvm_cmd_run_init(int argc, const char **argv)
+ static int kvm_cmd_run_work(struct kvm *kvm)
+ {
+ 	int i;
+-	void *ret = NULL;
++	int exit_status = 0;
+ 
+ 	for (i = 0; i < kvm->nrcpus; i++) {
+ 		if (pthread_create(&kvm->cpus[i]->thread, NULL, kvm_cpu_thread, kvm->cpus[i]) != 0)
+@@ -638,7 +638,20 @@ static int kvm_cmd_run_work(struct kvm *kvm)
+ 	}
+ 
+ 	/* Only VCPU #0 is going to exit by itself when shutting down */
+-	return pthread_join(kvm->cpus[0]->thread, &ret);
++	/* */
++	/* BUT It's not actually true for reboot sequence  */
++	/* Reboot is sending SIGKVMEXIT signal to all cpus */
++	/* and that is causing race condition on removing  */
++	/* devices (which are pausing exiting cpus).       */
++	/* This piece of code waits for death of all       */
++	/* cpu threads   				   */
++	for (i = 0; i < kvm->nrcpus; i++) {
++		int retval;
++		retval = pthread_join(kvm->cpus[i]->thread, NULL);
++		if (retval > exit_status)
++			exit_status=retval;
```

Comment:
add a comment, that you are going to return "biggest" error - or just exit loop and return error on first error occurred

but better if you can, intercept the value returned by thread loop (original *ret) and check error returned pthread_join and because return value from this function is propagated up to main function, we could spot the error after running lkvm and wrapped by rkt up to systemd - "Errors should never pass silently." (zen of python)